core/esmf-aspect-model-urn/src/main/java/org/eclipse/esmf/aspectmodel/urn/AspectModelUrn.java

```diff
@@ -352,6 +352,12 @@ public boolean isSammUrn() {
       return isSammUrn;
    }
 
+   public AspectModelUrn withName(String name) {
```

Comment:
* Please add JavaDoc for this method
* If you want to create another AspectModelUrn with another local name but the same prefix, there is no need for the `substring()` call, because `getUrnPrefix()` already does this:
```java
return fromUrn( getUrnPrefix() + name );
```